tests/src/gui/testqgsrelationreferencewidget.cpp

```diff
@@ -408,12 +406,21 @@ void TestQgsRelationReferenceWidget::testAddEntry()
       bool saveEdits( QgsVectorLayer * ) const override {return true;};
   };
 
+  QgsAdvancedDigitizingDockWidget cadDockWidget( &canvas );
   QgsAttributeEditorContext context;
   DummyVectorLayerTools tools;
   context.setVectorLayerTools( &tools );
-  w.setEditorContext( context, nullptr, nullptr );
+  context.setCadDockWidget( &cadDockWidget );
+  w.setEditorContext( context, &canvas, nullptr );
   w.addEntry();
 
+  QgsFeature feat( mLayer1->fields() );
+  feat.setAttribute( QStringLiteral( "pk" ), 13 );
+  feat.setAttribute( QStringLiteral( "material" ), QStringLiteral( "steel" ) );
+  feat.setAttribute( QStringLiteral( "diameter" ), 140 );
+  feat.setAttribute( QStringLiteral( "raccord" ), "collar" );
+  w.mMapToolDigitize->digitized( feat );
```

Comment:
Not completely sure: shouldn't that be a new test instead of modifying an existing one?

Reply:
Actually, the old behavior is not existing anymore except for referenced layer with no geometry. And, it was not working. I fixed it and add a test for layer with no geometry. c4e7cf1